test/syscalls/call.js

```diff
@@ -0,0 +1,793 @@
+const debug = require('debug')
+const assert = require('assert')
+
+const Kernel = artifacts.require('./Kernel.sol')
+const abi = require('ethereumjs-abi')
+
+const beakerlib = require("../../beakerlib");
+
+// Valid Contracts
+const Valid = {
+    Adder: artifacts.require('test/valid/Adder.sol'),
+    Multiply: artifacts.require('test/valid/Multiply.sol'),
+    Divide: artifacts.require('test/valid/Divide.sol'),
+    SysCallTest: artifacts.require('test/valid/SysCallTest.sol'),
+    SysCallTestCall: artifacts.require('test/valid/SysCallTestCall.sol'),
+    FirstNestedCall: artifacts.require('test/valid/NestedCalls/FirstNestedCall.sol'),
+    SecondNestedCall: artifacts.require('test/valid/NestedCalls/SecondNestedCall.sol'),
+    ThirdNestedCall: artifacts.require('test/valid/NestedCalls/ThirdNestedCall.sol'),
+    FourthNestedCall: artifacts.require('test/valid/NestedCalls/FourthNestedCall.sol'),
+    FifthNestedCall: artifacts.require('test/valid/NestedCalls/FifthNestedCall.sol'),
+    SixthNestedCall: artifacts.require('test/valid/NestedCalls/SixthNestedCall.sol'),
+}
+
+const TestWrite = artifacts.require('test/TestWrite.sol');
+
+const Invalid = {
+    Simple: artifacts.require('test/invalid/Simple.sol')
+}
+
+contract('Kernel', function (accounts) {
+    describe('Call capability', function () {
+        const procName = "SysCallTestCall";
+        const bytecode = Valid.SysCallTestCall.bytecode;
+
+        describe('A()', function () {
```

Comment:
This test is a little misleading, since `TestWrite` isn't a valid procedure and isn't affected by the kernel. Due to the fact that `Kernel#createProcedure` doesn't do validation.

Reply:
I needed to test what was happening in the simplest possible way, and this was the simplest possible way, otherwise I'd needed to be testing nested calls before testing single calls.

Reply:
That makes sense, I would rename the test though so that's it's clear what the test is doing, for future reading.

Reply:
Done.